### PR TITLE
Better support for the pipeline.

### PR DESCRIPTION
**Pull request by LGDoor**
Add two steps in the pipeline syntax: `azureDl` for download and `azureUpload` for upload.

Example:
> azureDl downloadType: [value:'container', containerName: 'xxx'], storageCredentialId: 'xxx', includeFilesPattern: 'file1'

> azureUpload storageCredentialId: 'xxx', containerName: 'xxx', filesPath: 'file1'

### CODE REVIEW

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java [hunk 1]
```diff
@@ -306,6 +365,7 @@ public AzureStorageBuilderDesc getDescriptor() {
     }
 
     @Extension
+    @Symbol("azureDl")
```
Comment (chenkennt) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java]: azureDl -> azureDownload?

**Comment by clguimanMSFT** [comment]
I recommend adding this new functionality in the CI's sanity checks: https://github.com/azure-devops/ci/blob/master/vars/runSanityChecks.groovy

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java [hunk 2]
```diff
@@ -53,48 +53,94 @@
...
+    }
+
+    @DataBoundSetter
+    public void setIncludeArchiveZips(boolean includeArchiveZips) {
+        this.includeArchiveZips = includeArchiveZips;
+    }
+
     public static class DownloadType {
 
         public final String type;
```
Comment (LGDoor) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java]: It's used to bind the config value by the DataBoundSetter annotation processor

Comment (chenkennt) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java]: I mean should this be a private final field with a public getter? I see other files follow this pattern.

**Comment by LGDoor** [comment]
> I recommend adding this new functionality in the CI's sanity checks: https://github.com/azure-devops/ci/blob/master/vars/runSanityChecks.groovy
doing this after migrating the jenkins server

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java (hunk 2 above)
Comment (LGDoor) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java]: this is the recommended way by the [pipeline dev guide](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#constructor-vs-setters) to reduce the mandatory params and give appropriate default value.

**Pull request merged**